src/components/cards/BorrowAmountCard.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+
+import { div, mul } from 'biggystring'
+import * as React from 'react'
+import { View } from 'react-native'
+import FastImage from 'react-native-fast-image'
+
+import { DECIMAL_PRECISION } from '../../util/utils'
+import { type Theme, cacheStyles, useTheme } from '../services/ThemeContext.js'
+import { EdgeText } from '../themed/EdgeText.js'
+import { Card } from './Card'
+
+export const BorrowAmountCard = ({
+  dAppIcon,
+  fiatCode,
+  formattedAmount,
+  maxAmount,
+  title
+}: {
+  dAppIcon: string,
+  fiatCode: string,
+  formattedAmount: string,
+  maxAmount: string,
+  title: string
+}) => {
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  return (
+    <Card>
+      <View style={styles.cardContainer}>
+        <View style={styles.leftContainer}>
+          <EdgeText style={{ fontFamily: theme.fontFaceMedium }}>{title}</EdgeText>
+          <View style={styles.valueContainer}>
+            <EdgeText style={styles.valueFont}>{formattedAmount}</EdgeText>
+            <EdgeText>{fiatCode}</EdgeText>
+          </View>
+          <View style={[styles.bar, { width: mul(div(formattedAmount, maxAmount, DECIMAL_PRECISION), '100') + '%' }]} />
```

Comment:
Instead of having to calculated the width, you can use the [onLayout](https://reactnative.dev/docs/view#onlayout) prop on the 'valueContainer' View to get the text's width dynamically at run time and use that to se the bar's width.
We have some examples in our code on how to use it.